src/environment.rs

```diff
@@ -1828,12 +1828,29 @@ impl<'p> GroupedEnvironment<'p> {
         }
     }
 
+    pub fn platforms(&self) -> HashSet<Platform> {
+        match self {
+            GroupedEnvironment::Group(group) => group
+                .environments()
+                .flat_map(|env| env.platforms())
+                .collect(),
+            GroupedEnvironment::Environment(env) => env.platforms(),
+        }
+    }
+
     pub fn has_pypi_dependencies(&self) -> bool {
         match self {
             GroupedEnvironment::Group(group) => group.has_pypi_dependencies(),
             GroupedEnvironment::Environment(env) => env.has_pypi_dependencies(),
         }
     }
+
+    pub fn features(&self) -> Box<dyn Iterator<Item = &'p manifest::Feature> + 'p> {
+        match self {
+            GroupedEnvironment::Group(group) => Box::new(group.features(true)),
+            GroupedEnvironment::Environment(env) => Box::new(env.features(true)),
+        }
+    }
```

Comment:
Instead of boxing you can use `Either`.

Reply:
Now using either but it's still weird to me haha